src/main/java/edu/cornell/kfs/fp/service/impl/ScheduledAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+
+            Date entryDate = new Date(calendarCalculator.getTimeInMillis());
+            addToDateAndAmount(datesAndAmounts, scheduledAccountingLine, entryDate, totalAmount, endDate);
+
+            currentStartDate = new Date(calendarCalculator.getTimeInMillis());
+            totalAmount = totalAmount.subtract(scheduledAccountingLine.getPartialAmount());
+
+            iterationCount++;
+        }
+
+        determineAccountingLineErrors(scheduledAccountingLine, rowId, iterationCount);
+
+        return datesAndAmounts;
+    }
+
+    private boolean validDateForLooping(Date endDate, Date currentStartDate) {
+        return currentStartDate.before(endDate) || KfsDateUtils.isSameDay(currentStartDate, endDate);
+    }
+
+    private void determineAccountingLineErrors(ScheduledSourceAccountingLine scheduledAccountingLine, int rowId, int iterationCount) {
+        String errorField = DOCUMENT_ERROR_PREFIX + "sourceAccountingLine[" + rowId + "].partialTransactionCount";
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("scheduledAccountingLine.getPartialTransactionCount() : " + scheduledAccountingLine.getPartialTransactionCount() 
+                + "  iterationCount: " + iterationCount);
+        }
+        if (isAccountingLineTotalMoreThanRecurranceTotal(scheduledAccountingLine)) {
+            GlobalVariables.getMessageMap().putError(errorField, CUKFSKeyConstants.ERROR_RCDV_RECURRNECE_SUM_LESS_THAN_TANSACTION);
+        } else if (Integer.parseInt(scheduledAccountingLine.getPartialTransactionCount()) > iterationCount) {
+            GlobalVariables.getMessageMap().putError(errorField, CUKFSKeyConstants.ERROR_RCDV_TOO_MANY_RECURRENCES);
+        }
+    }
+    
+    private boolean isAccountingLineTotalMoreThanRecurranceTotal(ScheduledSourceAccountingLine scheduledAccountingLine) {
+        KualiDecimal totalAmount = calculateTotalAmount(scheduledAccountingLine.getAmount());
+        double recurranceTotal = scheduledAccountingLine.getPartialAmount().doubleValue() * Integer.parseInt(scheduledAccountingLine.getPartialTransactionCount());
+        boolean isAccountingLineTotalMoreThanRecurranceTotal = totalAmount.doubleValue() > recurranceTotal;
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("totalAmount = " + totalAmount + "  recurranceTotal: " + recurranceTotal 
+                + "  scheduledAccountingLine.getPartialTransactionCount(): " + scheduledAccountingLine.getPartialTransactionCount() 
+                + "  isAccountingLineTotalMoreThanRecurranceTotal: " + (isAccountingLineTotalMoreThanRecurranceTotal));
+        }
+        return isAccountingLineTotalMoreThanRecurranceTotal;
+    }
+
+    private void addToDateAndAmount(TreeMap<Date, KualiDecimal> datesAndAmounts, ScheduledSourceAccountingLine scheduledAccountingLine, 
+            Date entryDate, KualiDecimal totalAmountLeft, Date endDate) {
+        if (!entryDate.after(endDate)) {
```

Comment:
is this another place that should be before, or before && same to avoid the negation?